motionplan/rrt.go

```diff
@@ -1,21 +1,90 @@
 package motionplan
 
+import (
+	"context"
+
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
 const (
 	// Number of planner iterations before giving up.
 	defaultPlanIter = 2000
 )
 
+type rrtParallelPlanner interface {
+	motionPlanner
+	rrtBackgroundRunner(context.Context, spatialmath.Pose, []referenceframe.Input, *rrtParallelPlannerShared)
+}
+
+type rrtParallelPlannerShared struct {
```

Comment:
This would be easy to rework in the future if/when those planners exist.

Possibly this should simply anonymously wrap `plannerOptions`?

Without those other algorithms existing, it's hard to think about.

Reply:
I think the thing I'm implying is that the options should be tied to the instance of the motion planner class rather than being passed as an argument into the background manager function.  There are a lot of functions that are passed in the options as an argument and it would simplify things if we could access it as part of the mp struct

Reply:
And to add to this, why can't we just put the rest of this struct in the motion planner struct?

Reply:
I feel that another question is do we eventually want `plannerOptions` to incorporate options for 2D/3D motion planning alongside all this other stuff? Like a one-stop-shop, even if some motion planning isn't ever going to use certain options.

Kind of sounds like Peter's suggestion about anonymously wrapping a generic `plannerOptions` would cut down on the excess.

Reply:
Let's leave this for a second, follow-up PR though; this one is already gigantic, and having a smaller one just for that will allow us to think about it more easily

Reply:
good with me!